src/deadline/client/_pid_utils.py

```diff
@@ -49,27 +49,67 @@ def check_and_obtain_pid_lock_if_available(
                 f.close()  # Close the file before over-writing it
 
                 # Create a new pid file with the current process id
-                return _obtain_pid_lock_atomically(pid_file_full_path, logger)
+                return _obtain_pid_lock_atomically(
+                    pid_file_full_path, logger, int(current_process_pid)
+                )
 
     except Exception:
         # We already checked the file exists before reading it.
         # For any other unexpected exceptions, we should not delete the pid file and raise an error.
         raise
 
 
-def _obtain_pid_lock_atomically(pid_file_full_path: str, logger: ClickLogger) -> bool:
+def release_pid_lock(pid_file_location: str, current_process_pid: str, logger: ClickLogger) -> bool:
+    """
+    Releases the pid lock by deleting the pid file.
+    :param current_process_pid: current process's id
+    :param pid_file_location: location of the pid lock file
+    :param logger: Click logger component
+    :return:
+    """
+    logger.echo(f"Releasing pid lock at {pid_file_location}")
+
+    # Get the full path of the pid file at download progress location
+    pid_file_full_path = os.path.join(pid_file_location, PID_FILE_NAME)
+
+    # Check if pid lock file does not exist.
```

Comment:
This comment states exactly what the next line of code does. When writing comments like this, it's good to think about what information to communicate to a reader beyond what's immediately visible. In this case, it could be the reason why returning True in this case is the right thing to do.